graph/src/hypercert-minter-v-0.ts

```diff
@@ -0,0 +1,153 @@
+  Upgraded,
+  RightAdded,
+  WorkScopeAdded,
+  ImpactScopeAdded,
+} from "../generated/HypercertMinterV0/HypercertMinterV0";
+import { Hypercert, HypercertBalance, ImpactScope, Right, WorkScope } from "../generated/schema";
+
+export function handleAdminChanged(event: AdminChanged): void {
+  // Entities can be loaded from the store using a string ID; this ID
+  // needs to be unique across all entities of the same type
+  // let entity = ExampleEntity.load(event.transaction.from.toHex())
+  // // Entities only exist after they have been saved to the store;
+  // // `null` checks allow to create entities on demand
+  // if (!entity) {
+  //   entity = new ExampleEntity(event.transaction.from.toHex())
+  //   // Entity fields can be set using simple assignments
+  //   entity.count = BigInt.fromI32(0)
+  // }
+  // // BigInt and BigDecimal math are supported
+  // entity.count = entity.count + BigInt.fromI32(1)
+  // // Entity fields can be set based on event parameters
+  // entity.previousAdmin = event.params.previousAdmin
+  // entity.newAdmin = event.params.newAdmin
+  // // Entities can be written to the store with `.save()`
+  // entity.save()
+  // Note: If a handler doesn't require existing field values, it is faster
+  // _not_ to load the entity from the store. Instead, create it fresh with
+  // `new Entity(...)`, set the fields that should be updated and save the
+  // entity back to the store. Fields that were not set or unset remain
+  // unchanged, allowing for partial updates to be applied.
+  // It is also possible to access smart contracts from mappings. For
+  // example, the contract that has emitted the event can be connected to
+  // with:
+  //
+  // let contract = Contract.bind(event.address)
+  //
+  // The following functions can then be called on this contract to access
+  // state variables and other data:
+  //
+  // - contract.DEFAULT_ADMIN_ROLE(...)
+  // - contract.UPGRADER_ROLE(...)
+  // - contract.balanceOf(...)
+  // - contract.balanceOfBatch(...)
+  // - contract.contributorImpacts(...)
+  // - contract.counter(...)
+  // - contract.exists(...)
+  // - contract.getImpactCert(...)
+  // - contract.getRoleAdmin(...)
+  // - contract.hasRole(...)
+  // - contract.impactScopes(...)
+  // - contract.isApprovedForAll(...)
+  // - contract.proxiableUUID(...)
+  // - contract.rights(...)
+  // - contract.supportsInterface(...)
+  // - contract.totalSupply(...)
+  // - contract.uri(...)
+  // - contract.version(...)
+  // - contract.workScopes(...)
+}
+
+export function handleApprovalForAll(event: ApprovalForAll): void {}
+
+export function handleBeaconUpgraded(event: BeaconUpgraded): void {}
+
+export function handleImpactClaimed(event: ImpactClaimed): void {
+  let entity = new Hypercert(event.params.id.toString());
+  entity.claimHash = event.params.claimHash;
+  const contributors = [] as string[];
+  for (let i=0; i<event.params.contributors.length; i++)
+  {
+    const address = event.params.contributors[i];
```

Comment:
Is this enough for the current needs? We could also make contributors entities and get a many-to-many relationship